tools/webdriver/webdriver/client.py

```diff
@@ -425,7 +429,26 @@ def send_command(self, method, url, body=None):
             an error.
         """
         response = self.transport.send(method, url, body)
-        value = response.body["value"]
+
+        if "value" in response.body:
+            value = response.body["value"]
+        elif "error" in response.body:
+            logger.warning("'value' key not found in response but 'error' key "
+                           "found in root. This is a compliance issue, but the tests are "
+                           "working around it.")
+            value = response.body
```

Comment:
geckodriver implemented a lot of the spec before it got the new session handling right. I generally dislike this idea that people shouldn't be able to make incremental fixes  in whatever order they choose to prioritise but should be forced to implement certain things to even get tests to run.